code/lib/csf-tools/src/enrichCsf.test.ts

```diff
@@ -0,0 +1,170 @@
+/// <reference types="@types/jest" />;
+/* eslint-disable no-underscore-dangle */
+
+import { dedent } from 'ts-dedent';
+import { loadCsf, formatCsf } from './CsfFile';
+import { enrichCsf, extractSource } from './enrichCsf';
+
+expect.addSnapshotSerializer({
+  print: (val: any) => val,
+  test: (val) => true,
+});
+
+const enrich = (code: string) => {
+  // we don't actually care about the title
+
+  const csf = loadCsf(code, { makeTitle: (userTitle) => userTitle || 'default' }).parse();
+  enrichCsf(csf);
+  return formatCsf(csf);
+};
+
+describe('enrichCsf', () => {
+  describe('source', () => {
+    it('csf1', () => {
+      expect(
+        enrich(dedent`
+          export default {
+           title: 'Button',
+          }
+          export const Basic = () => <Button />
+        `)
+      ).toMatchInlineSnapshot(`
+        export default {
+          title: 'Button'
+        };
+        export const Basic = () => <Button />;
+        Basic.parameters = {
+          storySource: {
+            source: "() => <Button />"
+          },
+          ...Basic.parameters
+        };
+      `);
+    });
+    it('csf2', () => {
+      expect(
+        enrich(dedent`
+          export default {
+            title: 'Button',
+          }
+          const Template = (args) => <Button {...args} />
+          export const Basic = Template.bind({});
+          Basic.parameters = { foo: 'bar' }
+        `)
+      ).toMatchInlineSnapshot(`
+        export default {
+          title: 'Button'
+        };
+        const Template = args => <Button {...args} />;
+        export const Basic = Template.bind({});
+        Basic.parameters = {
+          foo: 'bar'
+        };
+        Basic.parameters = {
+          storySource: {
+            source: "args => <Button {...args} />"
```

Comment:
Does this mean every story ends up with this fairly unhelpful source (tells you the component, but basically nothing else?)

Reply:
Yes, this is basically for CSF1-style stories. We've added dynamically-generated snippets for most of the renderers, but users can still opt out if they want.